server/src/main/java/org/opensearch/identity/NativePrincipal.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.identity;
+
+import java.security.Principal;
+
+/**
+ * Available OpenSearch internal principals
+ *
+ * @opensearch.experimental
+ */
+public class NativePrincipal {
```

Comment:
`InternalPrincipal` may be? Do we actually need it? `NamedPrincipal` looks to be sufficient ...

Reply:
InternalPrincipal sounds good to me. I could remove this in favor of only using `NamedPrincipal` with a few caveats. Basically, the `InternalPrincipal` (here NativePrincipal), is used for handling the NoopSubject class. This class is in turn used for handling scenarios where the IdentityPlugin does not enforce any authc/z requirements. 

The reason the `NamedPrincipal` is not used for both scenarios is because the `NamedPrincipal` implements the Java `Principal` interface which expects things like `hashCode()` etc. to be overwritten. I could of course makes these null in the Noop scenario if that is preferred.

What sounds best to you?

Reply:
> What sounds best to you?

Your suggestion to remove it sounds good - I think we could always introduce additional principals later on but it seems like this one is not needed (and to your point, it is not a principal per se - it does not implement `Principal`), thank you

Reply:
I went ahead and moved the internal principal code into the `NamedPrincipal` class as just a special instance that has a static constant.